scripts/solr_builder/tests/test_fn_to_cli.py

```diff
@@ -35,9 +35,7 @@ def test_parse_docs(self):
 
     def test_type_to_argparse(self):
         assert FnToCLI.type_to_argparse(int) == {'type': int}
-        assert FnToCLI.type_to_argparse(typing.Optional[int]) == {  # noqa: UP007
-            'type': int
-        }
+        assert FnToCLI.type_to_argparse(int | None) == {'type': int}
```

Comment:
Let's keep both of these around. This code is meant to support parsing types, so supporting the Optional types seems useful.